File: 0623-add-one-row-to-tree/0623-add-one-row-to-tree.java
public class Solution {
    public TreeNode addOneRow(TreeNode root, int val, int depth) {
        var newRoot = new TreeNode(0, root, null);
        var currDepth = 0;
        Queue<TreeNode> queue = new LinkedList<>();

        queue.add(newRoot);
        while(queue.size() > 0){
            currDepth++;
            var count = queue.size();
            
            for(var i=0; i<count; i++){
                var node = queue.poll();
                if(depth == currDepth){
                    node.left = new TreeNode(val, node.left, null);
                    node.right = new TreeNode(val, null, node.right);
                }
                else{
                    if(node.left != null) queue.add(node.left);
                    if(node.right != null) queue.add(node.right);
                }
            }
        }

        return newRoot.left;
    }
}

File: 0623-add-one-row-to-tree/0623-add-one-row-to-tree.py
class Solution:
    def addOneRow(self, root: Optional[TreeNode], val: int, depth: int) -> Optional[TreeNode]:
        new_root = TreeNode(0, root)
        curr_depth = 0
        queue = deque()

        queue.append(new_root)
        while queue:
            curr_depth += 1
            count = len(queue)
            
            for i in range(count):
                node = queue.popleft()
                if depth == curr_depth:
                    node.left = TreeNode(val, node.left, None)
                    node.right = TreeNode(val, None, node.right)
                else:
                    if node.left != None: queue.append(node.left)
                    if node.right != None: queue.append(node.right)

        return new_root.left